thriftbp/server_middlewares.go

```diff
@@ -380,3 +383,59 @@ func RecoverPanic(name string, next thrift.TProcessorFunction) thrift.TProcessor
 		},
 	}
 }
+
+// PrometheusMetrics tracks Prometheus metrics specific to the Thrift service.
+func PrometheusMetrics(service string) thrift.ProcessorMiddleware {
```

Comment:
nit: we might want to rename `service` arg to `slug` to be more consistent with other code.